src/material/material.c

```diff
@@ -42,6 +42,8 @@ void	material_free(t_material *material)
 
 int	material_deep_copy(t_material *dst, t_material *src)
 {
+	if (dst == src)
+		return (0);
```

Comment:
Added this in case we accidentally call this function with the destination and source being the same t_material.